src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -129,28 +130,29 @@ protected boolean calculateIfDirty() {
 
             // protocol come first
             if (hdrLen > 0) {
-                bab.append(HPUB_SP_BYTES);
-            } else {
-                bab.append(PUB_SP_BYTES);
+                bab.append(HPUB_SP_BYTES, 0, HPUB_SP_BYTES_LEN);
+            }
+            else {
+                bab.append(PUB_SP_BYTES, 0, PUB_SP_BYTES_LEN);
             }
 
             // next comes the subject
-            bab.append(subject, utf8mode ? UTF_8 : US_ASCII).append(SP);
+            bab.append(subject.getBytes(UTF_8)).append(SP);
 
             // reply to if it's there
             if (replyToLen > 0) {
-                bab.append(replyTo).append(SP);
+                bab.append(replyTo.getBytes(UTF_8)).append(SP);
             }
 
             // header length if there are headers
             if (hdrLen > 0) {
-                bab.append(Integer.toString(hdrLen)).append(SP);
+                bab.append(Integer.toString(hdrLen).getBytes(US_ASCII)).append(SP);

```

Comment:
the `subject` and `replyTo` use `UTF_8`, while the header and payload use `US_ASCII`
not quite sure, but shouldn't it be either utf8 or ascii?

Reply:
Numbers will never be anything other than ascii, so I don't have to worry that a character in the string of numbers could be utf8. Here is an example protocol line.  
```
HPUB SUBJECT REPLY 48 55␍␊NATS/1.0␍␊Header1: X␍␊Header1: Y␍␊Header2: Z␍␊␍␊PAYLOAD␍␊
```

This line of code could have been left alone, but the old version called append with a string, which checked what the default encoding of the bab and got those bytes, then made another call to the append that takes bytes. I just reduced the amount of code being executed by doing more manually.

Reply:
Ah, didn't recognise them just being numbers, thanks :)